include/aikido/perception/DetectedObject.hpp

```diff
@@ -30,37 +31,53 @@ namespace perception {
 /// }
 /// \endcode
 
+/// ObjectName may be different from \c AssetDatabase key, which may be
+/// a higher-level description of the object.
+/// For example, objectName can be "cantaloupe" while \c AssetDatabase key
+/// is "food_item".
+
 class DetectedObject
 {
 public:
-  /// Construct a \c DetectedObject
-  /// \param[in] uid Unique ID for object in Aikido world. Same UID -> Same
-  /// Object
-  /// \param[in] assetKey Key for AssetDatabase passed into constructor of
-  /// PoseEstimatorModule. Defines visuals / assets.
+  /// Default constructor.
+  DetectedObject() = default;
+
+  /// Constructs a \c DetectedObject.
+  /// Object's Uid is creqted as objectName_objectID
+  /// \param[in] objectName Name of the object.
+  /// \param[in] objectId Unique ID for object in Aikido world.
   /// \param[in] detectionFrameID Frame ID from ROS Marker
-  /// \param[in] yamlStr String of additional parameters for object. Can
-  /// override objAssetDBKey by specifying "db_key".
+  /// \param[in] yamlStr YAML string which specifies \c AssetDatabase
+  /// with {"db_key": key} and additional parameters.
   DetectedObject(
-      const std::string& uid,
-      const std::string& assetKey,
+      const std::string& objectName,
+      const int& objectId,
```

Comment:
Any reason for `int` vs `std::size_t`?

Reply:
While that should be generally the case, this is just whatever [visualization_msgs/Marker](http://docs.ros.org/melodic/api/visualization_msgs/html/msg/Marker.html) assigns as marker id, which is `int32`. I do not have a strong reason to enforce that marker id to be nonnegative yet, so I'd like to keep it this way.